### PR TITLE
Ensure `PUBLISH_DIR` exists

### LINKED ISSUE #41: `PUBLISH_DIR` might not exist
The `PUBLISH_DIR` constant is always defined. However it is possible that it targets a directory that's not been created yet. There are several ways to handle it, including:
  - creating it as an empty directory
  - making it default to `process.cwd()`
  - throwing an error. That's probably not the best solution since it breaks builds.

This plugin should probably create it as an empty directory when it does not exist. Otherwise, this plugin fails with:

```
Failed to write immutable cache headers
ENOENT: no such file or directory, open '/opt/build/repo/public/_headers' 
    Error: Failed to write immutable cache headers 
    ENOENT: no such file or directory, open '/opt/build/repo/public/_headers' 
```

### PR DESCRIPTION
**Pull request by ehmicky**
Fixed #41.

### CODE REVIEW

**Comment by Munter** [comment]
How can `PUBLISH_DIR` not exist in the `postBuild` hook? Is this a case of the user not having generated any output at all in their build command?

It does indeed seem useless to even run the plugin when there is nothing to run it on

**Comment by Munter** [comment]
I'm sure you're going to have this same problem in almost any of the other plugins that run `postBuild`. Should netlify-build maybe ensure that at least an empty directory exists?

**Comment by ehmicky** [comment]
It's a little complicated unfortunately to ensure this. I have opened a discussion in https://github.com/netlify/build/issues/1331, feel free to drop in!

**Comment by Munter** [comment]
Well, even if you end up creating the directory, the addition in this PR will not throw, and an empty result from the glob patterns in the subfont configuration should just skip any work, so I think itøs safe to merge

File: lib/index.js [hunk 1]
```diff
@@ -25,6 +26,8 @@ module.exports = {
       `Running subfont version ${require('subfont/package.json').version}`
     );
 
+    await makeDir(PUBLISH_DIR)
```
Comment (Munter) [inline lib/index.js:29]: Linting failure

```suggestion
    await makeDir(PUBLISH_DIR);
```

**Pull request merged**